src/js/stores/MesosSummaryStore.js

```diff
@@ -186,17 +199,31 @@ class MesosSummaryStore extends GetSetBaseStore {
     if (!Array.isArray(data)) {
       return MesosSummaryActions.fetchSummary(TimeScales.MINUTE);
     }
+
+    // If we get less data than the history length
+    // fill the front with the `n` copies of the earliest snapshot available
+    if (data.length < Config.historyLength) {
+      let diff = Config.historyLength - data.length;
+      for (var i = 0; i < diff; i++) {
+        data.unshift(Util.deepCopy(data[0]));
+      }
+    }
+
     // Multiply Config.stateRefresh in order to use larger time slices
     data = MesosSummaryUtil.addTimestampsToData(data, Config.getRefreshRate());
     data.forEach((datum) => {
       this.processSummary(datum, {silent: true});
     });
+    this.set({lastRequestTime: Date.now()});
```

Comment:
I think we could run into animation issues here because our interval isn't accurate - we effectively have `Config.getRefreshRate() + requestResponseTime`